cpp/src/arrow/compute/kernels/vector_selection_internal.cc

```diff
@@ -60,6 +60,7 @@ void RegisterSelectionFunction(const std::string& name, FunctionDoc doc,
         {std::move(kernel_data.value_type), std::move(kernel_data.selection_type)},
         OutputType(FirstType));
     base_kernel.exec = kernel_data.exec;
+    base_kernel.exec_chunked = kernel_data.chunked_exec;
```

Comment:
The member variable is called `exec_chunked` but the type is called `ChunkedExec` (so confusing). In this PR I ended up sticking to `chunked_exec`. Once everything is reviewed and merged I could try to unify things to the direction people prefer.